spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
Shouldn't we do both these operation with the same `session`?
I mean something like:
```
protected void handleMessageInternal(final Message<?> message) throws Exception {
	final RemoteFileTemplate<F> remoteFileTemplate = this.remoteFileTemplate;
	remoteFileTemplate.execute(new SessionCallbackWithoutResult<F>() {

		@Override
		protected void doInSessionWithoutResult(Session<F> session) throws IOException {
			String path = remoteFileTemplate.send(message, FileTransferringMessageHandler.this.mode);
			if (FileTransferringMessageHandler.this.chmod != null && isChmodCapable()) {
				doChmod(remoteFileTemplate, path, FileTransferringMessageHandler.this.chmod);
			}
		}

	});
}
```

But right: our `remoteFileTemplate.execute()` should support `ThreadLocal` propagation for the `Session` object do not obtain a new one from the `CF`.

We may address that from the separate JIRA of course.
Right now it is just for "heads up!".